news/11647-allowlist_channels

```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Rename `whitelist_channels` to `allowlist_channels`. (#11647)
```

Comment:
I might reword this to say that we're changing any instances of `whitelist` to `allowlist` and `blacklist` to `denylist`

Reply:
Hehe, I explicitly wrote this with the opposite mindset; outline the actual change, not the policy/objective that resulted in the change. I wrote it this way since the change only resulted in one meaningful variable change. 🤷🏻 

I have no strong opinion here so if others agree that this should be reworded I'll make the change.

Reply:
I think this is fine as written, it's not worth adding troll traps into the changelog. The issues and pull requests have more details than whatever we reasonably would have to write in the changelog to explain the policy change about naming things.